### PR TITLE
Add Gelman-Rubin statistics and effective sample size

### PR DESCRIPTION
**Pull request by AustinRochford**

### CODE REVIEW

File: webmc3/stats.py [hunk 1]
```diff
@@ -7,3 +9,25 @@ def autocorr(trace, varname, max_lag=100):
...
     ])
+
+
+def effective_n(trace, varname):
+    return effective_n_all(trace)[varname]
+
+
+# adding an LRU cache is a bit of a hack to memorize the effective sample size
+# which may be slow to calculate for complex models
+@lru_cache(maxsize=1)
```
Comment (ColCarroll) [inline webmc3/stats.py:22]: drive-by suggestion of using a class and lazy properties like https://stackoverflow.com/a/6849299 ?

Comment (AustinRochford) [reply webmc3/stats.py:22]: 🤔 what would be the practical difference? I'm hesitant to roll my own caching.

Comment (ColCarroll) [reply webmc3/stats.py:22]: Good point – i'm not sure there's a practical difference. I think I find object oriented code more intuitive for caching like this: "this object holds data about this trace", instead of "this function might remember this trace". (love following along on this, btw!)

Comment (AustinRochford) [reply webmc3/stats.py:22]: Makes sense. Ever since I learned Haskell I really don't like OO Python, so it may be a matter of taste.

Glad you're enjoying my code!
